src/controller/include/controller/recording_controller.h

```diff
@@ -0,0 +1,91 @@
+#include "controller/controller_interface.h"
+
+#include <array>
+#include <bitset>
+#include <fstream>
+#include <jsoncpp/json/json.h>
+#include <jsoncpp/json/writer.h>
+
+#pragma once
+
+namespace controller {
+class RecordingController : public ControllerInterface {
```

Comment:
There's a lot of repeated code here between `RecordingController` and `Controller`.

Can we make `RecordingController` inherit from `Controller` instead? Then, we can mark only the methods that need to be changed as `virtual` and simply override them in `RecordingController`

Reply:
Yup, sounds great! Tested it out, Works Perfectly!